c++/src/kj/compat/http.c++

```diff
@@ -3714,6 +4011,91 @@ public:
     });
   }
 
+  kj::Promise<ConnectResponse> connect(kj::StringPtr host, const HttpHeaders& headers) override {
+    KJ_REQUIRE(!upgraded,
+        "can't make further requests on this HttpClient because it has been or is in the process "
+        "of being upgraded");
+    KJ_REQUIRE(!closed,
+        "this HttpClient's connection has been closed by the server or due to an error");
+    KJ_REQUIRE(httpOutput.canReuse(),
+        "can't start new request until previous request body has been fully written");
+
+    closeWatcherTask = nullptr;
+
+    // Mark upgraded for now even though the tunnel could fail, because we can't allow pipelined
+    // requests in the meantime.
+    upgraded = true;
+
+    kj::StringPtr connectionHeaders[HttpHeaders::CONNECTION_HEADERS_COUNT];
+
+    httpOutput.writeHeaders(headers.serializeConnectRequest(host, connectionHeaders));
+
+    auto id = ++counter;
+
+    return httpInput.readResponseHeaders().then(
+        [this,id](HttpHeaders::ResponseOrProtocolError&& responseOrProtocolError)
+            -> HttpClient::ConnectResponse {
+      KJ_SWITCH_ONEOF(responseOrProtocolError) {
+        KJ_CASE_ONEOF(response, HttpHeaders::Response) {
+          auto& responseHeaders = httpInput.getHeaders();
+          if (response.statusCode < 200 || response.statusCode >= 300) {
+            // Any statusCode that is not in the 2xx range in interpreted
+            // as an HTTP response. Any status code in the 2xx range is
+            // interpreted as a successful CONNECT response.
+            closed = true;
+            return HttpClient::ConnectResponse {
+              .statusCode = response.statusCode,
```

Comment:
Unfortunately, designated initializers are a C++20 feature, and we still target C++14 in this repo. This is why the MSVC build is failing. (GCC and Clang supported this syntax before it was officially in the standard.)